src/grpc_client.cc

```diff
@@ -23,6 +25,48 @@ namespace grpc_labview
 
     //---------------------------------------------------------------------
     //---------------------------------------------------------------------
+    static bool ichar_equals(char a, char b)
+    {
+        return std::tolower(static_cast<unsigned char>(a)) == std::tolower(static_cast<unsigned char>(b));
+    }
+
+    static bool IsLoopbackAddress(const char* address)
+    {
+        std::string hostname(address);
+
+        std::regex hostFromUriRegex(R"(^\w+://([^/?#:]+))");
+        std::smatch match;
+
+        if (std::regex_search(hostname, match, hostFromUriRegex) && match.size() > 1)
+        {
+            hostname = match.str(1); // The host part is in the first capture group
+        }
+        else
+        {
+            size_t lastColonPosition = hostname.rfind(':');
+            if (lastColonPosition != std::string::npos)
+            {
+                hostname = hostname.substr(0, lastColonPosition);
+            }
+        }
+
+        std::string localhost("localhost");
+        bool isLoopback = hostname.size() == localhost.size() && std::equal(hostname.begin(), hostname.end(), localhost.begin(), ichar_equals);
```

Comment:
This is a roundabout way to compare strings. If you're going to use the current locale for case folding, you might as well use [strcasecmp](https://www.man7.org/linux/man-pages/man3/strcasecmp.3.html). Calling `tolower` on each letter may do the wrong thing in some cases, such as [German ß](https://en.wikipedia.org/wiki/%C3%9F)).

Reply:
I'm rusty on C++ and maybe I'm being dense, but `strcasecmp` isn't part of the C++ standard library is it. At least when I tried to use it, I got errors saying "error C1083: Cannot open include file: 'strings.h': No such file or
 directory". I don't care about locales in this case since I only care about matching against this one string. Maybe trying to ignore case here is going overboard?

Reply:
Oops, `strcasecmp` is a BSD function. The Microsoft equivalent is `_stricmp`. If you search NI's codesearch for these, you will find a lot of #ifdefs.

> I don't care about locales in this case since I only care about matching against this one string. Maybe trying to ignore case here is going overboard?

Comparing to localhost without using the name resolver is kind of a hack anyway. 

It would be easier to ignore SpOnGeBoB cAsE and do `hostname == "localhost" || hostname == "LOCALHOST"` :)

Reply:
Fair enough. My goal is to make this "just work" for what I would expect to be the majority of cases where loopback communication is used. If use of DNS host names for the localhost proves to be a support issue, we can revisit and improve this further. I'd rather not add the overhead of a DNS lookup to every connection at this point.